tools/cluster_mgr.py

```diff
@@ -153,11 +149,180 @@ def main():
+                if slot_range["end"] + 1 == other_slot_range["start"]:
+                    slot_range["end"] = other_slot_range["end"]
+                    slot_ranges.remove(other_slot_range)
+                    break
+
+    for slot in range(args.slot_start, args.slot_end+1):
+        shard, slot_range = find_slot(slot, config)
+        if shard == None:
+            continue
+        if shard == new_owner:
+            continue
+        remove_slot(slot, slot_range, shard)
+        add_slot(slot, new_owner)
+
+    for shard in config:
+        pack(shard["slot_ranges"])
+
+    print(f'Pushing new config:\n{json.dumps(config, indent=2)}\n')
+    push_config(config)
+
+
+def print_config(args):
+    config = build_config_from_existing(args)
+    print(json.dumps(config, indent=2))
+
+
+def destroy(args):
+    config = build_config_from_existing(args)
+    for node in get_nodes_from_config(config):
+        send_command(node, ["shutdown"])
+
+
+def main():
+    parser = argparse.ArgumentParser(description='''
+Local Cluster Manager
+
+Example usage:
+
+Create a 3+3 nodes cluster:
+  ./cluster_mgr.py --action=create --num_masters=3 --replicas_per_master=1
+
+Connect to cluster and print current config:
+  ./cluster_mgr.py --action=print
+
+Connect to cluster and shutdown all nodes:
+  ./cluster_mgr.py --action=shutdown
+
+Connect to cluster and move slots 10-20 to master with port 7002:
+  ./cluster_mgr.py --action=move --slot_start=10 --slot_end=20 --new_owner=7002
+''')
+    parser.add_argument('--action', default='',
+                        help='Which action to take? create: start a new instance, move=move slots')
+    parser.add_argument('--num_masters', type=int, default=3,
+                        help='Number of master nodes in cluster')
+    parser.add_argument('--replicas_per_master', type=int, default=0,
+                        help='How many replicas for each master')
+    parser.add_argument('--first_port', type=int,
+                        default=7001, help="First master's port")
+    parser.add_argument('--threads', type=int, default=2,
+                        help="Threads per node")
+    parser.add_argument('--slot_start', type=int,
+                        default=0, help="First slot to move (inclusive)")
+    parser.add_argument('--slot_end', type=int,
+                        default=100, help="Last slot to move (inclusive)")
+    parser.add_argument('--target_port', type=int, default=0,
+                        help="Master port to take ownership over slots in range "
+                        "[--slot_start, --slot_end]")
+    args = parser.parse_args()
+
+    if (args.action == 'create'):
+        create(args)
+    elif (args.action == 'shutdown'):
+        destroy(args)
+    elif (args.action == 'move'):
+        move(args)
+    elif (args.action == 'print'):
+        print_config(args)
+    else:
+        print(f'Error - unknown action "{args.action}"')
+        exit(-1)
```

Comment:
If you have more, you can create a dict[str -> function] and call it by lowercased action